etna/analysis/plotters.py

```diff
@@ -116,51 +189,69 @@ def plot_forecast(
         else:
             plot_df = pd.DataFrame(columns=["timestamp", "target", "segment"])
 
-        segment_forecast_df = forecast_ts[:, segment, :][segment]
-
         if (train_ts is not None) and (n_train_samples != 0):
             ax[i].plot(plot_df.index.values, plot_df.target.values, label="train")
         if test_ts is not None:
             ax[i].plot(segment_test_df.index.values, segment_test_df.target.values, color="purple", label="test")
-        ax[i].plot(segment_forecast_df.index.values, segment_forecast_df.target.values, color="r", label="forecast")
-
-        if prediction_intervals and quantiles is not None:
-            alpha = np.linspace(0, 1, len(quantiles) // 2 + 2)[1:-1]
-            for quantile in range(len(quantiles) // 2):
-                values_low = segment_forecast_df["target_" + str(quantiles[quantile])].values
-                values_high = segment_forecast_df["target_" + str(quantiles[-quantile - 1])].values
-                if quantile == len(quantiles) // 2 - 1:
-                    ax[i].fill_between(
-                        segment_forecast_df.index.values,
-                        values_low,
-                        values_high,
-                        facecolor="g",
-                        alpha=alpha[quantile],
-                        label=f"{quantiles[quantile]}-{quantiles[-quantile-1]} prediction interval",
-                    )
-                else:
-                    values_next = segment_forecast_df["target_" + str(quantiles[quantile + 1])].values
-                    ax[i].fill_between(
+
+        # plot forecast plot for each of given forecasts
+        colors = plt.cm.Set2.colors
+        for j, (forecast_name, forecast) in enumerate(forecast_results.items()):
+            legend_prefix = f"{forecast_name}: " if num_forecasts > 1 else ""
+            forecast_color = colors[j % len(colors)]
+
+            segment_forecast_df = forecast[:, segment, :][segment].sort_values(by="timestamp")
+            ax[i].plot(
+                segment_forecast_df.index.values,
+                segment_forecast_df.target.values,
+                color=forecast_color,
+                linewidth=1,
+                label=f"{legend_prefix}forecast",
+            )
+
+            # draw prediction intervals from outer layers to inner ones
+            if prediction_intervals and quantiles is not None:
+                alpha = np.linspace(0, 1 / 2, len(quantiles) // 2 + 2)[1:-1]
+                for quantile in range(len(quantiles) // 2):
```

Comment:
looks like it is not `quantile` but `quantile_idx`

Reply:
This part of code wasn't written by me.
Ok, I can fix it.

Reply:
thanks!